### PR TITLE
Separation power for Pions and Electrons

### PR DESCRIPTION
**Pull request by HorstMa**
This PR adds the separation power that can be trended for pions and electrons at the MIP point.

### CODE REVIEW

File: Detectors/TPC/qc/include/TPCQC/PID.h [hunk 1]
```diff
@@ -88,6 +88,7 @@ class PID
   }
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TH1>>>& getMapOfHisto() { return mMapHist; }
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TCanvas>>>& getMapOfCanvas() { return mMapCanvas; }
+  TCanvas* getSeparationPower() { return cSeparationPowerCanvas; }
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]:
```suggestion
  TCanvas* getSeparationPowerCanvas() { return mSeparationPowerCanvas; }
```

File: Detectors/TPC/qc/include/TPCQC/PID.h [hunk 2]
```diff
@@ -107,6 +108,8 @@ class PID
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TCanvas>>> mMapCanvas;
   // Map for Histograms which will be put onto the canvases, and not published separately
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TH1>>> mMapHistCanvas;
+  // Canvas for Trending Separation Power
+  TCanvas* cSeparationPowerCanvas;
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]:
```suggestion
  TCanvas* mSeparationPowerCanvas = nullptr;
```

File: Detectors/TPC/qc/src/PID.cxx [hunk 3]
```diff
@@ -21,6 +21,7 @@
 #include "TCanvas.h"
 #include "TMathBase.h"
 #include "TObjArray.h"
+#include "TF1.h"
```
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]: Ist this header used?
```suggestion
```

File: Detectors/TPC/qc/src/PID.cxx [hunk 4]
```diff
@@ -102,6 +103,7 @@ void PID::initializeHistograms()
     mMapCanvas["CdEdxPIDHypothesisVsp"].emplace_back(std::make_unique<TCanvas>("CdEdxPIDHypothesisVsp", "PID Hypothesis Ratio"));
     mMapCanvas["CdEdxPIDHypothesisVsp"].at(0)->Divide(5, 2);
   }
+  cSeparationPowerCanvas = new TCanvas("CSeparationPower", "Separation Power");
```
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]:
```suggestion
  mSeparationPowerCanvas = new TCanvas("CSeparationPower", "Separation Power");
```

**Review by wiechula — CHANGES_REQUESTED** [review]

**Review by shahor02 — CHANGES_REQUESTED** [review]
@HorstMa please rebase your branch to fresh dev and force push.

File: Detectors/TPC/qc/include/TPCQC/PID.h [hunk 5]
```diff
@@ -88,6 +88,7 @@ class PID
   }
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TH1>>>& getMapOfHisto() { return mMapHist; }
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TCanvas>>>& getMapOfCanvas() { return mMapCanvas; }
+  TCanvas* getSeparationPowerCanvas() { return mSeparationPowerCanvas; }
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]:
```suggestion
  TCanvas* getSeparationPowerCanvas() { return mSeparationPowerCanvas.get(); }
```

File: Detectors/TPC/qc/include/TPCQC/PID.h [hunk 6]
```diff
@@ -107,6 +108,8 @@ class PID
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TCanvas>>> mMapCanvas;
   // Map for Histograms which will be put onto the canvases, and not published separately
   std::unordered_map<std::string_view, std::vector<std::unique_ptr<TH1>>> mMapHistCanvas;
+  // Canvas for Trending Separation Power
+  TCanvas* mSeparationPowerCanvas = nullptr;
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/PID.h]:
```suggestion
  std::unique_ptr<TCanvas> mSeparationPowerCanvas;
```

File: Detectors/TPC/qc/src/PID.cxx [hunk 7]
```diff
@@ -102,6 +102,7 @@ void PID::initializeHistograms()
     mMapCanvas["CdEdxPIDHypothesisVsp"].emplace_back(std::make_unique<TCanvas>("CdEdxPIDHypothesisVsp", "PID Hypothesis Ratio"));
     mMapCanvas["CdEdxPIDHypothesisVsp"].at(0)->Divide(5, 2);
   }
+  mSeparationPowerCanvas = new TCanvas("CSeparationPower", "Separation Power");
```
Comment (wiechula) [inline Detectors/TPC/qc/src/PID.cxx]:
```suggestion
  mSeparationPowerCanvas.reset(new TCanvas("CSeparationPower", "Separation Power"));
```

**Review by wiechula — CHANGES_REQUESTED** [review]

**Review by wiechula — APPROVED** [review]

**Pull request merged**